sdk/maps/maps-route-rest/samples-dev/route.ts

```diff
@@ -0,0 +1,207 @@
+   *
+   * More info is available at https://docs.microsoft.com/en-us/azure/azure-maps/azure-maps-authentication.
+   */
+  /** Shared Key authentication (subscription-key) */
+  const subscriptionKey = process.env.MAPS_SUBSCRIPTION_KEY || "";
+  const credential = new AzureKeyCredential(subscriptionKey);
+  const client = createMapsRouteClient(credential);
+
+  /** Azure Active Directory (Azure AD) authentication */
+  // const credential = new DefaultAzureCredential();
+  // const mapsClientId = process.env.MAPS_CLIENT_ID || "";
+  // const client = createMapsRouteClient(credential, mapsClientId);
+
+  console.log(" --- Get route directions:");
+  const getRouteDirectionsResult = await client.path("/route/directions/{format}", "json").get({
+    queryParameters: {
+      query: toColonDelimitedLatLonString([
+        [51.368752, -0.118332],
+        [41.385426, -0.128929],
+      ]),
+      vehicleWidthInMeters: 2,
+      vehicleHeightInMeters: 2,
+      vehicleLoadType: "USHazmatClass1",
+      travelMode: "truck",
+      isCommercialVehicle: true,
+    },
+  });
+
+  if (isUnexpected(getRouteDirectionsResult)) {
+    throw getRouteDirectionsResult.body.error;
+  }
+  console.log(getRouteDirectionsResult.body);
+
+  console.log(" --- Get route directions with additional parameters:");
+  const routeDirectionsWithParamResult = await client
+    .path("/route/directions/{format}", "json")
+    .post({
+      queryParameters: {
+        query: toColonDelimitedLatLonString([
+          [52.50931, 13.42936],
+          [52.50274, 13.43872],
+        ]),
+      },
+      body: {
+        supportingPoints: {
+          type: "GeometryCollection",
+          geometries: [
+            {
+              type: "Point",
+              coordinates: [13.42936, 52.5093],
+            },
+            {
+              type: "Point",
+              coordinates: [13.42859, 52.50844],
+            },
+          ],
+        },
+        avoidVignette: ["AUS", "CHE"],
+        avoidAreas: {
+          type: "MultiPolygon",
+          coordinates: [
+            [
+              [
+                [-122.39456176757811, 47.489368981370724],
+                [-122.00454711914061, 47.489368981370724],
+                [-122.00454711914061, 47.65151268066222],
+                [-122.39456176757811, 47.65151268066222],
+                [-122.39456176757811, 47.489368981370724],
+              ],
+            ],
+            [
+              [
+                [100.0, 0.0],
+                [101.0, 0.0],
+                [101.0, 1.0],
+                [100.0, 1.0],
+                [100.0, 0.0],
+              ],
+            ],
+          ],
+        },
+      },
+    });
+
+  if (isUnexpected(routeDirectionsWithParamResult)) {
+    throw routeDirectionsWithParamResult.body.error;
+  }
+  console.log(routeDirectionsWithParamResult.body);
+
+  console.log(" --- Get route range:");
+  const routeRangeResult = await client
```

Comment:
It seems like this could be split into its own sample file

Reply:
Are you suggesting separating different paths into different sample files?

Reply:
In .NET SDK, we separate the samples to route range, route direction, and route matrix.
https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/maps/Azure.Maps.Routing/samples

Reply:
I see. I'll separate the samples to align with .NET. Thanks!

Reply:
Done in 8154f233127a5f626d3428bb309e922665d41b76